DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/SwiftUI/Image+Drawing.swift

```diff
@@ -0,0 +1,128 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+#if os(iOS)
+
+import CoreGraphics
+import Foundation
+import UIKit
+
+@available(iOS 13.0, tvOS 13.0, *)
+extension GraphicsImage {
+    init?(rasterizing drawingContents: NSObject, origin: CGPoint) {
+        let rasterizationScale = UIScreen.main.scale
+
+        guard let cgImage = Self.drawingRasterizer.image(
+            rasterizing: drawingContents,
+            origin: origin,
+            scale: rasterizationScale
+        ) else {
+            return nil
+        }
+
+        self.init(contents: .cgImage(cgImage), scale: rasterizationScale, orientation: .up)
+    }
+}
+
+@available(iOS 13.0, tvOS 13.0, *)
+extension GraphicsImage {
+    fileprivate class DrawingRasterizer: @unchecked Sendable {
+        private let cache = NSCache<NSNumber, CGImage>()
+        private let lock = NSRecursiveLock()
+
+        init() {
+            cache.countLimit = 20
+        }
+
+        func image(rasterizing drawingContents: NSObject, origin: CGPoint, scale: CGFloat) -> CGImage? {
+            lock.lock()
+            defer { lock.unlock() }
+
+            let imageKey = drawingContents.hash as NSNumber
+
+            if let image = cache.object(forKey: imageKey) {
+                return image
+            }
+
+            guard let image = CGImage.image(
+                rasterizing: drawingContents,
+                origin: origin,
+                scale: scale
+            ) else {
+                return nil
+            }
+
+            cache.setObject(image, forKey: imageKey)
+            return image
+        }
+    }
+
+    fileprivate static let drawingRasterizer = DrawingRasterizer()
+}
+
+extension CGImage {
+    @available(iOS 13.0, tvOS 13.0, *)
+    fileprivate static func image(rasterizing drawingContents: NSObject, origin: CGPoint, scale: CGFloat) -> CGImage? {
+        guard
+            let cls = DrawingContents.cls,
+            type(of: drawingContents).isSubclass(of: cls),
+            drawingContents.responds(to: DrawingContents.renderInContext),
+            let bounds = drawingContents.value(forKey: DrawingContents.bounds) as? CGRect
+        else {
+            return nil
+        }
+
+        // Compute image size
+
+        let width = Int((bounds.width + 1.5) * scale)
+        let height = Int((bounds.height + 1.5) * scale)
+
+        // Create a grayscale, alpha-only bitmap context
+
+        guard let bitmapContext = CGContext(
+            data: nil,
+            width: width,
+            height: height,
+            bitsPerComponent: 8,
+            bytesPerRow: width,
+            space: CGColorSpaceCreateDeviceGray(),
+            bitmapInfo: CGImageAlphaInfo.alphaOnly.rawValue
+        ) else {
+            return nil
+        }
```

Comment:
We could also rasterize drawings into full color bitmaps, allowing [multi-color or palette symbols](https://developer.apple.com/documentation/swiftui/symbolrenderingmode). The problem is that, in some instances, `RBMovedDisplayListContents` data uses `white` color. For instance, when tinting symbols with an accent color, or just arbitrarily as I experienced during my tests.

I couldn't find a way to determine which bitmap format to choose, as I don't have any information about how the symbol is rendered on the screen.

Any ideas are welcome!